Fix checking of sudo's path in launch script.

Apteryks opened the pull request:
The sudo binary's existence in the PATH was being checked on the host machine rather than in the container.  Also, use the POSIX `command` utility, which should be even more prevalent than `which`.

joufella commented on cqfd:
```diff
@@ -263,10 +263,10 @@ make_launcher()
-useradd -s /bin/bash -ou $UID -g $GROUPS -d "$cqfd_user_home" $cqfd_user >/dev/null 2>&1
+
+groupadd -og $GROUPS -f builders
+useradd -s /bin/bash -ou $UID -g $GROUPS -d "$cqfd_user_home" $cqfd_user
```
As an UX improvement, what you could do, like you did for sudo, is checking for groupadd/useradd availability in the container image, and letting the user know that if they are not available, cqfd will not be able to work. In a second step, running the commands and trapping their failure, to let the user know cqfd can't work also.

Apteryks replied:
Implemented in commit 9e21970476bf59b723861e85aa8db18c40aa518c, thanks for the prompt review!

joufella commented on cqfd:
```diff
@@ -271,7 +271,7 @@ chown $UID:$GROUPS $cqfd_user_home
 # run the provided command in the working directory
 cd $cqfd_user_cwd
-if [ -x "$(which sudo)" ]; then
+if command -v sudo > /dev/null 2>&1; then
```
good catch!

Apteryks commented:
@joufella ping; anything remaining you'd like to see addressed for this PR?

joufella commented on cqfd:
```diff
@@ -265,13 +265,39 @@ make_launcher()
-useradd -s /bin/bash -ou $UID -g $GROUPS -d "$cqfd_user_home" $cqfd_user
-chown $UID:$GROUPS $cqfd_user_home
+HAS_SUDO=
+FAILED=
```
those empty declarations are not needed. Also note that stylewise, majority of local variables are not capitalized

Apteryks replied:
Done.

joufella commented on cqfd:
```diff
@@ -265,13 +265,39 @@ make_launcher()
+Docker container... Please refer to the above error messages."
+}
+
+check_requirements
```
if it's called once and sits just above, I see no value in making it a function, you can add a comment on the block

Apteryks replied:
Done.

joufella commented on cqfd:
```diff
@@ -265,13 +265,39 @@ make_launcher()
-cd $cqfd_user_cwd
-if command -v sudo > /dev/null 2>&1; then
+cd $cqfd_user_cwd || die "Changing directory to \"$cqfd_user_cwd\" failed."
+if [ -n "$HAS_SUDO" ]; then
```
should it be  $HAS_SUDO or \$HAS_SUDO ?

Apteryks replied:
\$has_sudo, good catch! I've made it possible to catch this in the tests.

Apteryks commented:
I've addressed your comments and made the tests better (the output was being hidden by grep -q, and we couldn't know if it was 'su' or 'sudo' that was being used.

joufella commented:
Thanks @Apteryks I merged your changes !

Apteryks commented:
@joufella thank you! Is there a new release on the horizon? At least one customer makes use of the build_context feature and has to use some random commit. It'd be great to have them use a proper release.
